Source/Core/ArcGISTiledElevationTerrainProvider.js

```diff
@@ -304,7 +304,14 @@ Object.defineProperties(ArcGISTiledElevationTerrainProvider.prototype, {
    */
   availability: {
     get: function () {
-      return undefined;
+      //>>includeStart('debug', pragmas.debug)
+      if (!this._ready) {
+        throw new DeveloperError(
+          "availability must not be called before the terrain provider is ready."
+        );
+      }
+      //>>includeEnd('debug');
+      return this._tilesAvailable;
```

Comment:
This was the other fix mentioned... I'm not really across how the `TileAvailability` class works and if this `_tilesAvailable ` property is setup correctly. I did notice there's actually two properties `_tilesAvailablityLoaded` and `_tilesAvailable` in the ArcGIS Terrain Provider; not sure which one is correct in this case.

Either way, I think this makes `sampleTerrainMostDetailed` work as expected 👍

Reply:
yeah, this seems like the right thing to return. I'm still wrapping my head around `_tilesAvailablityLoaded`. I think it only says whether availability has been loaded for tiles, not whether a tile is actually available. Not sure why `_tilesAvailable` can't do take care of this, but there's probably a good reason.